dev/tools/gen_keycodes/README.md

```diff
@@ -58,109 +58,62 @@ the API.
 However, if you are porting Flutter to a new platform, you should follow the
 following guidelines for specifying logical key codes.
 
-The logical key code is a 37-bit integer in a namespace that we control and
-define. It has values in the following ranges.
+The logical key code is a 52-bit integer (due to the limitation of JavaScript).
+The entire namespace is divided into 32-bit *planes*. The upper 20 bits of the
+ID represent the plane ID, while the lower 32 bits represent values in the
+plane.  For example, plane 0x1 refers to the range 0x1 0000 0000 -
+0x1 FFFF FFFF.  Each plane manages how the values within the range are assigned.
 
-- **0x00 0000 0000 - 0x0 0010 FFFF**: For keys that generate Unicode
+The planes are planned as follows:
+
+- **Plane 0x00**: The Unicode plane. This plane contains keys that generate Unicode
   characters when pressed (this includes dead keys, but not e.g. function keys
-  or shift keys), the logical key code is the Unicode code point corresponding
```

Comment:
I removed lots of text because I think they're more about the key model instead of definition to the key, which does not fit into this README quite well. The key model should be described in the documentation of `HardwareKeyboard` or a design doc instead. I'm not sure if/how they should be moved there though.